src/js/protocol_uri.js

```diff
@@ -0,0 +1,89 @@
+import $ from 'jquery';
+
+import GUI from './gui';
+import serial from './serial';
+import PortHandler from "./port_handler";
+
+export const protocol = 'btfltcp';
+export const protocolScheme = `${protocol}://`;
+
+export async function registerProtocolURI() {
+    if (GUI.isCordova()) {
+        handleCordovaOpenEvents();
+        return;
+    }
+
+    const command = `"${process.execPath}" $_URL_`;
+
+    console.log(`Registering ${protocolScheme} to '${command}'`);
+    try {
+        const ProtocolRegistry = require('protocol-registry');
```

Comment:
TODO: This dynamic `require` seems to prevent `rollup` from including `protocol-registry` in the build; as a result the NW.js builds don't properly register the schema.

Reply:
Final update: :relieved: I dropped `protocol-registry` in favor of implementing everything in our `gulpfile`. :) This is a much cleaner setup overall - the only slight drawback being that one has to install/re-install the configurator for the URI schema to register. Thus during local development the URI schema will resolve to the _installed_ configurator and not to the current development build. Of course (if needed) one can manually update their `.deskop`/`.plist`/registry to point to the development executable.